### ISSUE TITLE
Cpush always? returns 0

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#43]
**Reported by mchasal on 7 Jun 2002 15:32 UTC**
```
I executed a cpush while one of my nodes was
(intentionally) down and here are the results:

[/root](root@wallace)# cpush /etc/hosts
building file list ... done
wrote 44 bytes  read 20 bytes  42.67 bytes/sec
total size is 258  speedup is 4.03
Secure connection to oscarnode1.val.pok.ibm.com refused.
unexpected EOF in read_timeout
[/root](root@wallace)# echo $?
0

I also tried it with a bogus file that didn't exist and
it still returned 0, and it even tried to rsync.

c3 should return some indication that the push failed
so it can be used programmatically. It also may be nice
to check if the file exists on the source before
calling rsync.

I don't think this is a showstopper for 1.3, but should
be addressed.

```

**Comment by dikim33** [comment]
**Comment by mchasal on 7 Jun 2002 15:55 UTC**
```

Just another point, cexec seems to have the same issue:

[systemconfig](root@wallace)# cexec grep SIS /etc/hosts     
              
*************************  oscar_cluster 
*************************
processing node  oscarnode1.val.pok.ibm.com
processing node  oscarnode2.val.pok.ibm.com
Secure connection to oscarnode1.val.pok.ibm.com refused.
--------- oscarnode1.val.pok.ibm.com---------

--------- oscarnode2.val.pok.ibm.com---------
# These entries are managed by SIS.

[systemconfig](root@wallace)# echo $?
0

```

**Comment by dikim33** [comment]
**Comment by  on 7 Jun 2002 18:06 UTC**
```
Logged In: NO 

All the C3 commands will do this. C3 is, at this point,
"buyer beware", the only error checking is on the command
line. The next version will have a good deal of runtime
error checking. I'm currently working on it :)
```

**Comment by dikim33** [comment]
**Comment by bligneri on 23 Nov 2004 01:29 UTC**
```

Hello Thomas,

I was wondering who is in charge of the c3 tools at ORNL ?
If it is not you, can you assign this bug to the person in
charge. If the bug will not be fixed / is already fixed, can
you modify the tracker accordingly.
```

**Comment by dikim33** [comment]
**Comment by naughtont on 24 Nov 2004 21:09 UTC**
```

Not possible for 4.0, moving to 4.1
```

**Comment by dikim33** [comment]
**Comment by efocht on 2 Aug 2006 14:50 UTC**
If this problem still exists, it would be very useful to fix it NOW and not wait 2 more years...

**Comment by dikim33** [comment]
**Modified by jparpail on 11 Apr 2007 08:17 UTC**

**Comment by dikim33** [comment]
**Comment by wesbland on 16 Jul 2007 19:26 UTC**
This has been fixed for the next version of c3 (5.0) which will be released soon.
